dagshub/mlflow/patch.py

```diff
@@ -0,0 +1,176 @@
+                continue
+
+            if is_class and full_attr_name in self.classes_to_patch:
+                for class_attr_name in dir(attr):
+                    if class_attr_name.startswith("_"):
+                        continue
+                    class_attr = getattr(attr, class_attr_name)
+                    full_class_attr_name = f"{full_attr_name}.{class_attr_name}"
+                    if isinstance(class_attr, FunctionType) and not self.is_function_already_patched(
+                        full_class_attr_name
+                    ):
+                        self.original_func_lookup[full_class_attr_name] = class_attr
+                        setattr(attr, class_attr_name, self._wrap_func(class_attr))
+
+            is_module = isinstance(attr, ModuleType)
+            # Ignore lazy loaded modules, don't want to load them all at once
+            # TODO: maybe let users tune this somehow, although from what I've seen most of those are wrappers,
+            #  and the actual mlflow logging functions are eagerly loaded
+            is_lazy_module = isinstance(attr, LazyLoader)
+
+            if not is_module or is_lazy_module:
+                continue
+
+            full_module_name = attr.__name__
+
+            if not full_module_name.startswith("mlflow"):
+                continue
+
+            if full_module_name in self.patched_modules:
+                continue
+
+            # Recurse into the module and patch it
+            self.monkey_patch_module(attr)
+
+    def unpatch(self):
+        logger.warning("Unpatching MLflow to raise exceptions again")
+        global _global_mlflow_patch
+
+        for unpatched_key, unpatched_func in self.original_func_lookup.items():
+            attrs = unpatched_key.split(".")
+            module_name = attrs[0]
+            attrs = attrs[1:]
+            attr = sys.modules[module_name]
+            for index, attr_name in enumerate(attrs):
+                parent_attr = attr
+                attr = getattr(parent_attr, attr_name)
+                if index == len(attrs) - 1:
+                    setattr(parent_attr, attr_name, unpatched_func)
+
+        self.patched_modules.clear()
+        self.original_func_lookup.clear()
+        _global_mlflow_patch = None
+
+
+_global_mlflow_patch: Optional[MlflowMonkeyPatch] = None
+
+
+def patch_mlflow():
+    import mlflow
+
+    global _global_mlflow_patch
+
+    if _global_mlflow_patch is not None:
+        logger.warning("MLflow is already patched")
+        return
+
+    logger.warning(
+        "Patching MLflow to prevent any MLflow exceptions from being raised. "
+        "Call dagshub.mlflow.unpatch_mlflow() to undo"
+    )
+
+    patch = MlflowMonkeyPatch()
+    patch.monkey_patch_module(mlflow)
+    _global_mlflow_patch = patch
+
+
+def unpatch_mlflow():
+    if _global_mlflow_patch is None:
+        logger.warning("MLflow should be unpatched already")
+        return
+    _global_mlflow_patch.unpatch()
```

Comment:
Suggestion if it's not too difficult - could you add a context manager that temporarily turns off the patch while inside the context? If I have a specific important call. WDYT?
```python
with dagshub.mlflow.pause_patch():
   mlflow.log_param('super-important', 42)
```

Reply:
This is actually dangerous thread-wise, because I'm patching the whole mlflow module, so any parallel request will also have the patch unapplied.

If you still want this functionality, as an alternative I can add a kwarg to the wrapping function that disables the patch for this function.
```
mlflow.log_param(blabla, dont_ignore_exception=True)
```

The only problem is it kind of prevents code reuse + you have to duplicate the flag in however many calls to mlflow you do

Reply:
As we discussed, you could set a thread-local flag that causes the existing hooks to no-op while it's on.
But leaving that out of the current PR.